packages/plugin-meta-extractor/src/webpack-plugin.ts

```diff
@@ -0,0 +1,35 @@
+import path from 'node:path';
+import * as webpack from 'webpack';
+import { extractPluginMeta } from './meta-extractor.js';
+
+const DEFAULT_GENERATED_FILENAME = 'plugin.generated.json';
+
+export type GrafanaPluginMetaExtractorOptions = {
+  entryFile?: string;
+  outputAssetName?: string;
+};
+
+export class GrafanaPluginMetaExtractor {
+  options: GrafanaPluginMetaExtractorOptions;
+
+  constructor(options: GrafanaPluginMetaExtractorOptions) {
+    this.options = options;
```

Comment:
It might be worth using [schema-utils](https://github.com/webpack/schema-utils) to validate options being passed to the plugin.

Reply:
Yepp, that's a good suggestion! (Although we might don't need this in case we use the entries from the webpack config.)

Reply:
Fixed in https://github.com/grafana/plugin-tools/pull/805/commits/d8b0c6cfe13859e9d67520e369ce02d2ca818950.